cluster-autoscaler/utils/gpu/gpu.go

```diff
@@ -68,6 +68,21 @@ func FilterOutNodesWithUnreadyGpus(allNodes, readyNodes []*apiv1.Node) ([]*apiv1
 	return newAllNodes, newReadyNodes
 }
 
+// GetGpuType returns name of the GPU used on the node or empty string if there's no GPU
+// if the GPU type is unknown, "generic" is returned
+// NOTE: current implementation is GKE/GCE-specific
+func GetGpuType(node *apiv1.Node) string {
```

Comment:
Why add new method if we already have GetNodeTargetGpus? Especially if you already use output of GetNodeTargetGpus in one place and the output of this function in another place?

Reply:
TBH I just haven't noticed it.

But the logic behind both is a little bit different - here I just need GPU type and not count, I'd also prefer not to touch Cloud Provider since it's not needed for the name itself.

Additionally, if I understand correctly, GetNodeTargetGpus returns non-zero value only in GKE whereas I'd like the metrics to work everywhere, even if they are not as detailed.

Reply:
The problem is this won't work reliably. GPU capacity / allocatable is only set on node after the GPU driver is installed, so it's not a reliable way of telling if the underlying VM has a GPU.

Reply:
As discussed - I've switched the logic to use label first but still support "generic" GPU if the node has GPU capacity set.